src/outbox/outboxActions.js

```diff
@@ -121,12 +113,10 @@ const mapEmailsToUsers = (emails, allUsersByEmail, ownUser) => {
   return result;
 };
 
-// TODO type: `string | NamedUser[]` is a bit confusing.
-type DataFromNarrow = {|
-  type: 'private' | 'stream',
-  display_recipient: string | NamedUser[],
-  subject: string,
-|};
+type DataFromNarrow = SubsetProperties<
```

Comment:
> In fact, use our SubsetProperties type-helper to get
the real thing directly from the actual Outbox type so we don't
have to duplicate the definition.